kevm-pyk/src/kevm_pyk/foundry.py

```diff
@@ -684,8 +744,9 @@ def foundry_list(foundry_root: Path) -> list[str]:
 def foundry_remove_node(foundry_root: Path, test: str, node: NodeIdLike) -> None:
     foundry = Foundry(foundry_root)
     apr_proofs_dir = foundry.out / 'apr_proofs'
-    contract_name, test_name = test.split('.')
-    proof_digest = foundry.proof_digest(contract_name, test_name)
+    contract_name = test.split('.')[0]
+    test_sig = foundry.matching_sig(test).split('.')[1]
+    proof_digest = foundry.proof_digest(contract_name, test_sig)
```

Comment:
This logic seems to be duplicated a few times. Can it be extracted in a new function?

Reply:
Thanks!